File: JavaScript/aesGcmEncryption.js
const crypto = require("crypto");

function aesGcmEncryption(plainText, key) {
  /**
   * Generate a random initialization vector (IV)
   */
  const ivLength = 12;
  const iv = crypto.randomBytes(ivLength);

  /**
   * Create a secret key from the provided key
   */
  const keySpec = Buffer.from(key);

  const cipher = crypto.createCipheriv("aes-128-gcm", keySpec, iv);
  let cipherText = cipher.update(plainText, "utf8");
  cipherText = Buffer.concat([cipherText, cipher.final()]);

  /**
   * Combine the IV and cipher text into a single array
   */
  const tag = cipher.getAuthTag();
  const ivAndCipherText = Buffer.concat([iv, cipherText, tag]);

  /**
   * Encode the combined data as Base64
   */
  const base64Encoded = ivAndCipherText.toString("base64");

  return base64Encoded;
}

/**
 * ONLY FOR DEMO PURPOSE NEVER HARDCODE IN PRODUCTION
 */
const plainText = "ENTER THE TEXT";
const key = "YOUR SECRET KEY HERE";
const encryptedText = aesGcmEncryption(plainText, key);
console.log("Encrypted Text: " + encryptedText); // 👈 YOUR ENCRYPTED VALUE HERE

File: Swift/AesGcmEncryption.swift
import CryptoKit
import Foundation
import CommonCrypto

class AesGcmEncryption {
    func randomBytes(count: Int) -> Data {
        var data = Data(count: count)
        _ = data.withUnsafeMutableBytes { mutableBytes in
            SecRandomCopyBytes(kSecRandomDefault, count, mutableBytes.baseAddress!)
        }
        return data
    }

    func encryptGcmBase64(plainText: String, key: String) -> String? {
        /**
         * Generate a random initialization vector (IV)
         */
        let ivLength = 12
        let iv = randomBytes(count: ivLength)

        /**
         * Create a secret key from the provided key
         */
        guard let keyData = key.data(using: .utf8) else { return nil }
        let symmetricKey = SymmetricKey(data: keyData)

        let sealedBox: AES.GCM.SealedBox
        do {
            sealedBox = try AES.GCM.seal(plainText.data(using: .utf8)!, using: symmetricKey, nonce: AES.GCM.Nonce(data: iv))
        } catch {
            return nil
        }

        /**
         * Combine the IV and cipher text into a single array
         */
        let tag = sealedBox.tag
        let ivAndCipherText = iv + sealedBox.ciphertext + tag

        /**
         * Encode the IV and cipher text in Base64 format
         */
        let base64Encoded = ivAndCipherText.base64EncodedString()

        return base64Encoded
    }
}

/**
 * ONLY FOR DEMO PURPOSE NEVER HARDCODE IN PRODUCTION
 */
let aesGcmEncryption = AesGcmEncryption()
let plainText = "ENTER THE TEXT"
let key = "YOUR SECRET KEY HERE"
if let encryptedText = aesGcmEncryption.encryptGcmBase64(plainText: plainText, key: key) {
   print("Encrypted Text: " + encryptedText)
}